### PR TITLE
Add regression test guide

### PR DESCRIPTION
**Pull request by blouwern**
## Summary
This pull request is mainly created for adding a user/developer guide for running and extending regression tests in the MACESW project. And it makes minor improvements to error logging in several regression test macros. 

Fixes: #34 

## Type of change
- Documentation update

## Checklist before requesting review
- [x] I have read the [contributing guidelines](https://github.com/zhao-shihan/MACESW/blob/main/CONTRIBUTING.md)
- [x] The PR targets the `main` branch.
- [x] I linked related issues and provided context in the PR description.
- [x] My code follows the [coding style guide](https://github.com/zhao-shihan/MACESW/blob/main/STYLE_GUIDE.md) and linting rules.
- [x] I added/updated unit tests where applicable.
- [x] I updated relevant documentation (README, Doxygen, or design docs).
- [x] I ran the test-suite locally and all tests pass.
- [x] All CI checks pass.

## Implementation details (optional)
Documentation:

* Added a detailed `docs/regression-test-guide.md` that explains how to run regression tests locally, add new tests in CI, and describes the regression test workflow, conventions, and edge cases.
* Added a template and instructions to `src/test/scripts/ReadSomeDataTuple.cxx.template` for users to create new regression test macros for data tuples, including guidance on histogram definition and range settings.

Error reporting improvements (regression test macros):

* Updated all `Read*.cxx` macros (`ReadCDCSimHit.cxx`, `ReadECALSimHit.cxx`, `ReadMCPSimHit.cxx`, `ReadMMSSimTrack.cxx`, `ReadMRPCSimHit.cxx`, `ReadSimDecayVertex.cxx`, `ReadTTCSimHit.cxx`, `ReadVetoSimHit.cxx`) to print error messages to `std::cerr` instead of `std::cout` when failing to create output directories, improving error visibility and consistency. [[1]](diffhunk://#diff-23eacfa4b13d5eac934e66cc7444a5ea948165d1497038b2d90d618d4383dad3L26-R28) [[2]](diffhunk://#diff-fc5161c946eb77551bb5235eecb4fb74331679da68acc58827b86382e554f34aL26-R28) [[3]](diffhunk://#diff-3b9bf53957b8ba1457ec75b882fe5431acac7f5e82700fe1495a948edde3aa1eL26-R28) [[4]](diffhunk://#diff-1e5a5f61bc094ee31b797f83e126bafec1dfead204526d36b4900f92f75fde76L26-R28) [[5]](diffhunk://#diff-82d9912dba0a7aeab3738bb5bfc7fb934bcc774d5c26c2ef2893cf57c2cc76d1L26-R28) [[6]](diffhunk://#diff-bceaebbc89dae290cfb031cf69fbbb943dbcf52adb8463b3c47955c0a54b8e8dL26-R28) [[7]](diffhunk://#diff-580a38cd7d8bda1260e908410763d926cfb7a21f4a9d9787145379d942b84c53L26-R28) [[8]](diffhunk://#diff-d19f9ebae618f943d30947436bbdf2bed9920eb2010232779384354e507167fdL26-R28)

Minor cleanup:

* Removed a commented-out `main` function declaration in `TestCDCSimHit.cxx`.

## Reviewer notes
- Areas that need special attention during review: none
- Suggested reviewer: @zhao-shihan 

## Additional information
None

### CODE REVIEW

**Comment by zhao-shihan** [comment]
Once the PR is ready to review please request review from me (at the right panel on top).

File: src/test/scripts/TestSomeDataTuple.cxx.template [hunk 1]
```diff
@@ -0,0 +1,170 @@
+// Usage:
+// 1. Replace <data_tuple_name> with the name of datatuple (e.g. SimDecayVertex).
+// 2. Go to the "==USER DEFINE==" section and modify the dataframe definitions.
+// 3. Remove the ".template" tag of this file.
+// 4. Delete this guide text and the section guides, and format the code properly.
+
```
Comment (zhao-shihan) [inline src/test/scripts/TestSomeDataTuple.cxx.template]:
```suggestion
```

**Comment by zhao-shihan** [comment]
@blouwern Please let me know when the PR is ready to review.

**Comment by blouwern** [comment]
> [Apply these changes then it looks good to me.](https://github.com/zhao-shihan/MACESW/pull/44#pullrequestreview-3550835559)

@zhao-shihan The changes have now been completed.
